File: app/src/main/java/org/lifefortheorc/tudounotepad/model/NoteModel.java
package org.lifefortheorc.tudounotepad.model;

import android.os.Parcel;
import android.os.Parcelable;

import com.activeandroid.annotation.Column;
import com.activeandroid.annotation.Table;
import com.activeandroid.query.Delete;
import com.activeandroid.query.Select;

import java.util.List;

/**
 * Created by geminiwen on 15/7/2.
 */
@Table(name = "Note")
public class NoteModel extends BaseModel {

    @Column(name = "content")
    private String content;

    @Column(name = "time")
    private long time;

    public String getContent() {
        return content;
    }

    public void setContent(String content) {
        this.content = content;
    }

    public long getTime() {
        return time;
    }

    public void setTime(long time) {
        this.time = time;
    }

    public static List<NoteModel> queryNoteList() {
        return new Select().from(NoteModel.class).execute();
    }

    public static NoteModel queryById(long id) {
        return new Select().from(NoteModel.class)
                           .where("Id = ?", id).executeSingle();
    }

    public static NoteModel deleteById(long id) {
        return new Delete().from(NoteModel.class).where("Id = ?", id).executeSingle();
    }

    @Override
    public int describeContents() {
        return 0;
    }

    @Override
    public void writeToParcel(Parcel dest, int flags) {
        dest.writeString(this.content);
        dest.writeLong(this.time);
    }

    public NoteModel() {
    }

    protected NoteModel(Parcel in) {
        this.content = in.readString();
        this.time = in.readLong();
    }

    public static final Parcelable.Creator<NoteModel> CREATOR = new Parcelable.Creator<NoteModel>() {
        public NoteModel createFromParcel(Parcel source) {
            return new NoteModel(source);
        }

        public NoteModel[] newArray(int size) {
            return new NoteModel[size];
        }
    };
}

File: app/src/main/kotlin/org/lifefortheorc/tudounotepad/model/NoteModel.kt
package org.lifefortheorc.tudounotepad.model

import android.util.Log
import com.activeandroid.Model
import com.activeandroid.annotation.Column
import com.activeandroid.annotation.Table
import com.activeandroid.query.Delete
import com.activeandroid.query.Select
import kotlin.platform.platformStatic

/**
 * Created by geminiwen on 15/7/13.
 */
Table(name = "Note")
public data class NoteModel() : BaseModel(){

    Column(name="content")
    var content:String? = null

    Column(name="time")
    var time:Long? = null

    constructor(content:String, time:Long): this() {
        this.content = content
        this.time = time;
    }

    companion object {
        platformStatic fun queryNoteList(): List<NoteModel> {
            val modelList:List<NoteModel> = Select().from(javaClass<NoteModel>()).execute()
            return modelList
        }

        platformStatic fun queryById(id: Long): NoteModel {
            val model:NoteModel = Select().from(javaClass<NoteModel>()).where("Id = ?", id).executeSingle()
            return model;
        }

        platformStatic fun deleteById(id: Long): NoteModel {
            val model:NoteModel = Delete().from(javaClass<NoteModel>()).where("Id = ?", id).executeSingle()
            return model
        }
    }

}